scripts/check-lessons.sh

```
@@ -0,0 +1,29 @@
+rc=0
+dupes="$(ls "$LESSONS_DIR" | grep -oE '^lesson-[0-9]+' | sort | uniq -d || true)"
+if [ -n "$dupes" ]; then
+  echo "check-lessons: lesson numbers used more than once:"
+  for n in $dupes; do ls "$LESSONS_DIR" | grep -E "^${n}-" | sed 's/^/  /'; done
+  rc=1
+fi
+if [ -f "$INDEX" ]; then
+  for f in $(ls "$LESSONS_DIR" | grep -E '^lesson-[0-9]+.*\.md$'); do
+    grep -qF "$f" "$INDEX" || { echo "check-lessons: not in _index.md: $f"; rc=1; }
+  done
+  for f in $(grep -oE 'lesson-[0-9]+[A-Za-z0-9._-]*\.md' "$INDEX" | sort -u); do
```

Shellcheck (0.11.0)

[info] To read lines rather than words, pipe/redirect to a 'while read' loop.

(SC2013)